artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/MessageReferenceImpl.java

```diff
@@ -86,7 +86,7 @@ public int compare(MessageReference o1, MessageReference o2) {
 
    // Static --------------------------------------------------------
 
-   private static final int memoryOffset = 64;
+   private static final int memoryOffset = 72;
```

Comment:
Can we have some doc or something around this very magical number (its magical without any context), how do we derive 72?

Reply:
Good point, we have it already in other parts of the source code (see TypedProperties)

Reply:
Should it be a static constant then, where the constant is kept close to where if something changes the constant is. And a small javadoc regarding how it's calculated so that should something change people know how to recalc to update it

Reply:
> Should it be a static constant then, where the constant is kept close to where if something changes the constant is

The issue with TypedProperty is much subtle and complex to be fixed: I'm going to send a separate PR for that one

> And a small javadoc regarding how it's calculated so that should something change people know how to recalc to update it

+1, that's what I'm going to add